easybuild/easyblocks/s/scipion.py

```diff
@@ -0,0 +1,165 @@
+        """Custom configuration procedure for Scipion."""
+        # Setup scipion.conf, protocols.conf, and host.conf
+        # scipion.conf needs to be configured with paths to all the
+        # dependencies, stupid but that's the way it works at the
+        # moment.
+
+        # Create the config files and then make the subtitutions
+        cmd = './scipion --config %s/config/scipion.conf config' % self.installdir
+        run_cmd(cmd, log_all=True, simple=True)
+
+        # Mapping of EB software names and variables to patch in
+        # scipion.conf
+        map_sw2var_name = {
+            'ctffind': 'CTFFIND4_HOME',
+            'EMAN2': 'EMAN2DIR',
+            'frealign': 'FREALIGN_HOME',
+            'MotionCor2': 'MOTIONCOR2_HOME',
+            'RELION': 'RELION_HOME',
+            'Java': 'JAVA_HOME',
+            'VMD': 'VMD_HOME',
+            'MATLAB': 'MATLAB_DIR',
+        }
+
+        # Some other variables we need to change
+        regex_subs = [
+            (r'^CC = .*$', r'CC = %s' % os.getenv('CC')),
+            (r'^CXX = .*$', r'CXX = %s' % os.getenv('CXX')),
+            (r'^LINKERFORPROGRAMS = .*$', r'LINKERFORPROGRAMS = %s' % os.getenv('CXX')),
+            (r'^MPI_CC = .*$', r'MPI_CC = %s' % os.getenv('MPICC')),
+            (r'^MPI_CXX = .*$', r'MPI_CXX = %s' % os.getenv('MPICXX')),
+            (r'^MPI_LINKERFORPROGRAMS = .*$', r'MPI_LINKERFORPROGRAMS = %s' % os.getenv('MPICXX')),
+            (r'^MPI_LIBDIR = .*$', r'MPI_LIBDIR = /'),
+            (r'^MPI_INCLUDE = .*$', r'MPI_INCLUDE = /'),
+            (r'^MPI_BINDIR = .*$', r'MPI_BINDIR = /'),
+            # The EB motioncor2 build renames that binary to an explicit name
+            (r'^MOTIONCOR2_BIN = .*$', r'MOTIONCOR2_BIN = motioncor2'),
+        ]
+        for (k, v) in map_sw2var_name.items():
+            swroot = get_software_root(k)
+            if swroot:
+                regex_subs.extend([(r'^%s = .*$' % v, r'%s = %s' % (v, swroot))])
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            regex_subs.extend([(r'^(.*CUDA_LIB) = .*$', r'\1 = %s/lib64' % (cuda_root))])
+            regex_subs.extend([(r'^CUDA_BIN = .*$', r'CUDA_BIN = %s/bin' % (cuda_root))])
+            regex_subs.extend([(r'^CUDA = .*$', r'CUDA = True')])
+
+        java_root = get_software_root('Java')
+        if java_root:
+            jnipath = os.path.join(java_root, 'include')
+            jnipath = os.pathsep.join([jnipath, os.path.join(jnipath, 'linux')])
+            regex_subs.extend([(r'^JNI_CPPPATH = .*$', r'JNI_CPPPATH = %s' % jnipath)])
+
+        if get_software_root('OpenCV'):
+            opencv = 'True'
+        else:
+            opencv = 'False'
+        regex_subs.extend([(r'^OPENCV = .*$', r'OPENCV = %s' % opencv)])
+
+        if get_software_root('MATLAB'):
+            matlab = 'True'
+        else:
+            matlab = 'False'
+        regex_subs.extend([(r'^MATLAB = .*$', r'MATLAB = %s' % matlab)])
```

Comment:
please avoid duplicate code, and you can make this quite a bit shorter using `bool(...)`:

```python
for dep in ['MATLAB', 'OpenCV']:
    use_dep = bool(get_software_root(dep))
    regex_subs.append((r'^%s = .*$' % dep.upper(), r'%s = %s' % (dep.upper(), use_dep)))